### PR TITLE
include example for pulling source from github

### LINKED ISSUE #3906: Add example github url
Maybe commented out? This way we can make it easier for people to pull releases from GitHub.

### PR DESCRIPTION
**Pull request by CJ-Wright**
Closes #3906

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@conda-forge/staged-recipes @conda-forge/core thoughts?

**Comment by dougalsutherland** [comment]
I like the idea, but as-is it'll definitely lead to some confusion of people wondering what they should set the `org` variable to when they don't need it. Not sure what the best solution is....

**Comment by synapticarbors** [comment]
It might be helpful to give some explanation (either in the example recipe if concise or as a link to docs), why someone should choose the canonical pypi tarball vs the github archive. By having one commented out there is a pretty clear delineation of preference, but some clarification might help. 

I'm also not sure what the best approach is with the `org` variable other than to comment it out and then add a comment that when using a github archive, you need to uncomment that specific line.

**Comment by CJ-Wright** [comment]
I guess I could move the org `set` down into the body of the `source` next to the GH comments, but that could look ugly.

**Comment by CJ-Wright** [comment]
The discussion about pypi vs GH et al started about [here](https://gitter.im/conda-forge/conda-forge.github.io?at=5a3824afa2be466828b92bbf) on the gitter.

File: recipes/example/meta.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
 
 # Jinja variables help maintain the recipe as you'll update the version only here.
+{% set org = "simplejson" %}
```
Comment (scopatz) [inline recipes/example/meta.yaml]: I know that I have added org in some of my recipes, but in retrospect, it isn't something that changes often and it usually only in one or two parts of the code base.  I think it is better just remove it from the example.

File: recipes/example/meta.yaml [hunk 2]
```diff
@@ -16,6 +16,9 @@ package:
 source:
   fn: {{ name }}-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  # If getting the source from GitHub remove the line above
+  # uncomment the line below and modify as needed
+  # url: https://github.com/simplejson/{{ name }}/archive/{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/example/meta.yaml]: So the user/org part won't be correct generally. Should we template it? Note this obvious point?

Comment (CJ-Wright) [reply recipes/example/meta.yaml]: Templating seemed to be :-1: 
https://github.com/conda-forge/staged-recipes/pull/4734#issuecomment-355387903
https://github.com/conda-forge/staged-recipes/pull/4734#discussion_r160053925

**Comment by CJ-Wright** [comment]
@conda-forge/staged-recipes @conda-forge/core if this is good can I please get a merge?

**Pull request merged**